src/storage.rs

```diff
@@ -0,0 +1,115 @@
+// Key storage functionality for Clementine CLI
+
+use bitcoin::secp256k1::Keypair;
+use bitcoin::{Address, Network};
+use std::collections::HashMap;
+use std::fs;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+/// Store a keypair and its corresponding taproot address
+pub fn store_key(
+    keypair: &Keypair,
+    network: Network,
+    passphrase: Option<&str>,
+) -> Result<Address, Box<dyn std::error::Error>> {
+    // Check if passphrase encryption is requested
+    if passphrase.is_some() {
+        return Err("Passphrase encryption is not yet implemented".into());
+    }
+
+    // Calculate the taproot address for this keypair
+    let address = crate::bitcoin_utils::calculate_taproot_address(keypair, network);
+
+    // Create storage directory
+    let storage_dir = get_storage_dir()?;
+    fs::create_dir_all(&storage_dir)?;
+
+    // Store the keypair in plaintext (TODO: implement encryption)
+    let key_file = storage_dir.join(format!("key_{}.json", address));
+    let key_data = serde_json::json!({
+        "network": network.to_string(),
+        "address": address.to_string(),
+        "private_key": keypair.secret_key().display_secret().to_string(),
+        "public_key": keypair.public_key().to_string(),
+        "stored_at": chrono::Utc::now().to_rfc3339()
+    });
+    fs::write(key_file, serde_json::to_string_pretty(&key_data)?)?;
```

Comment:
can we chmod this file to 700 so that only the user who created it can read/write? this is standard for SSH keys and helps with security